### DISCUSSION TITLE
"avoid_negative_ts"

### DISCUSSION THREAD

**Discussion by TonyTTTT** [mifi/lossless-cut#1874] [General]
LosslessCut says  that "avoid_negative_ts"  "t's generally recommended to set this to one of: "auto", "make_zero""
However, the ffmpeg docs says "If you cut with stream copy (-c copy) you need to use the [-avoid_negative_ts 1](https://ffmpeg.org/ffmpeg-all.html#Format-Options) option if you want to use that segment with the [​concat demuxer](https://trac.ffmpeg.org/wiki/How%20to%20concatenate%20(join,%20merge)%20media%20files#demuxer) ."

Can you refer me to where its recommended to set to auto or make_zero?

**Comment by mifi** [comment]
So for reference, these are options with their numeric values and their corresponding string values ([source](https://github.com/FFmpeg/FFmpeg/blob/0c517fcbe8ee864c8390dd08696ec63621a4f3e0/libavformat/options_table.h#L98)):

- `1`/`make_non_negative` - Shift timestamps to make them non-negative. Also note that this affects only leading negative timestamps, and not non-monotonic negative timestamps.
- `2`/`make_zero` - Shift timestamps so that the first timestamp is 0.
- `-1`/`auto (default)` - Enables shifting when required by the target format.
- `0`/`disabled` - Disables shifting of timestamp.

your quote [comes from here](https://trac.ffmpeg.org/wiki/Seeking#Cuttingsmallsections)
> If you cut with stream copy (-c copy) you need to use the [-avoid_negative_ts 1](https://ffmpeg.org/ffmpeg-all.html#Format-Options) option if you want to use that segment with the [​concat demuxer](https://trac.ffmpeg.org/wiki/How%20to%20concatenate%20(join,%20merge)%20media%20files#demuxer).

so ffmpeg wiki is effectively saying that when using "merge/concat segments" in losslesscut, they recommend to use `-avoid_negative_ts make_non_negative`.

I don't have a source for losslesscut's  `generally recommended to set this to one of: "auto", "make_zero"`, but it was deduced through experience. However I think it indeed makes sense to update this text for the concat use case to respect ffmpeg's recommendation.
